code/modules/surgery/organs/liver.dm

```diff
@@ -107,16 +107,22 @@
 
 	if(filterToxins && !HAS_TRAIT(liver_owner, TRAIT_TOXINLOVER))
 		for(var/datum/reagent/toxin/toxin in cached_reagents)
-			var/amount = round(toxin.volume, CHEMICAL_QUANTISATION_LEVEL) // this is an optimization
-			if(belly)
-				amount += belly.reagents.get_reagent_amount(toxin.type)
-
-			// a 15u syringe is a nice baseline to scale lethality by
-			liver_damage += ((amount/15) * toxin.toxpwr) / liver_resistance
-
-			if(provide_pain_message != HAS_PAINFUL_TOXIN)
-				provide_pain_message = toxin.silent_toxin ? HAS_SILENT_TOXIN : HAS_PAINFUL_TOXIN
-
+			// Plasmamen don't take liver damage from plasma, but instead heal wounds
+			if(isplasmaman(liver_owner) && istype(toxin, /datum/reagent/toxin/plasma))
+				for(var/i in owner.all_wounds)
+					var/datum/wound/iter_wound = i
+					iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
+			else
+				var/amount = round(toxin.volume, CHEMICAL_QUANTISATION_LEVEL) // this is an optimization
+				if(belly)
+					amount += belly.reagents.get_reagent_amount(toxin.type)
+
+				// a 15u syringe is a nice baseline to scale lethality by
+				liver_damage += ((amount/15) * toxin.toxpwr) / liver_resistance
+
+				if(provide_pain_message != HAS_PAINFUL_TOXIN)
+					provide_pain_message = toxin.silent_toxin ? HAS_SILENT_TOXIN : HAS_PAINFUL_TOXIN
+	
```

Comment:
Rewrite this so that you use plasma lover liver trait instead of `isplasmaman`.  

It's also better to use a `continue` statement with the if loop so you don't have to nest your code deeper than it needs to be.

```
for(var/datum/reagent/toxin/toxin in cached_reagents)
	// Plasmamen don't take liver damage from plasma, but instead heal wounds
	if(isplasmaman(liver_owner) && istype(toxin, /datum/reagent/toxin/plasma))
		for(var/i in owner.all_wounds)
			var/datum/wound/iter_wound = i
			iter_wound.on_xadone(4 * REAGENTS_EFFECT_MULTIPLIER * delta_time) // plasmamen use plasma to reform their bones or whatever
		continue

	var/amount = round(toxin.volume, CHEMICAL_QUANTISATION_LEVEL) // this is an optimization
	if(belly)
		amount += belly.reagents.get_reagent_amount(toxin.type)

	// a 15u syringe is a nice baseline to scale lethality by
	liver_damage += ((amount/15) * toxin.toxpwr) / liver_resistance

	if(provide_pain_message != HAS_PAINFUL_TOXIN)
		provide_pain_message = toxin.silent_toxin ? HAS_SILENT_TOXIN : HAS_PAINFUL_TOXIN
```

Reply:
Gotcha -- good suggestions! I was not able to find a 'plasma lover liver trait' however, so I checked for the type of liver instead... If that's not an elegant enough solution I can change it, let me know if I missed something there.

There is one issue, and I'm going to put it back to draft for now about it---although this certainly fixes #72644 I am testing for #61424 and it seems that there is another hidden issue somewhere deeper in the code, unrelated to what I fixed.

The plasmaman I was trying to inject with a virus for testing was simply not getting the virus, when I checked its 'diseases' list in vars. Whereas the monkey I was testing on got it right away. I will have to look into this later today but if you have any ideas let me know.

Reply:
A followup to this, I found the issue! Everything seems to be working now. Also see #72609 for another plasmaman fix.